### ISSUE TITLE
Processing: Improve File Handling

### ISSUE THREAD

**Issue by Udayan12167** [coala-analyzer/coala#2456]
Right now when a new section has to be executed all the file contents that are loaded in the previous section are discarded and overlaps are reloaded. We would like to do away with this and reuse the already loaded file contents.

**Comment by Udayan12167** [comment]
#933

**Comment by ishanSrt** [comment]
Why? i think we even need to reload the file contents upon display of each result, since if the user is correcting the errors in the file ahead of the current result, coala still shows the errors in those lines.

**Comment by Makman2** [comment]
@ishanSrt we already have issues discussing re-analysis behaviour for such scenarios where the user has changed the file during analysis (can't find them atm). Currently imo it's not feasible, the user shall not do modifications while coala is running (otherwise they might be discarded). Consider you have global bears running which take super long, a change to a file retriggers the complete analysis. That would be very bad.

We need a more proper concept for this, until then we shouldn't reload files, this is really a bottleneck.

**Comment by ishanSrt** [comment]
But we ourselves, provide the option to open file in text editor.

**Comment by Makman2** [comment]
Good catch... Have you checked how it works? Not that we actually open a temporary copy of the file and perform some diff-magic so it works together with coala?

**Comment by ishanSrt** [comment]
wow! you really know each and every nook and corner of this project!
